onnxruntime/core/platform/windows/env.cc

```diff
@@ -124,52 +147,171 @@ class WindowsEnv : public Env {
   }
 
   int GetNumCpuCores() const override {
-    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
-    DWORD returnLength = sizeof(buffer);
-    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
-      // try GetSystemInfo
-      SYSTEM_INFO sysInfo;
-      GetSystemInfo(&sysInfo);
-      if (sysInfo.dwNumberOfProcessors <= 0) {
-        ORT_THROW("Fatal error: 0 count processors from GetSystemInfo");
+    int num_cores = 0;
+    for (const auto& group_info : group_info_vec_) {
+      num_cores += group_info.ActiveProcessorCount >> 1;
+    }
+    return num_cores;
+  }
+
+  size_t GetDefaultThreadpoolSetting(std::vector<size_t>& affinities) const override {
```

Comment:
General comment: One thing that isn't clear to me is whether the returned `affinities` vector is going to have a different representation/semantics across different platforms. For instance, I see here that the `affinities` vector is very closely related to the `KAFFINITY` structs on Windows. Will we be using the same representation on Linux? Or will it be a completely different representation, with a completely different interpretation?

Reply:
That's right. with completely different interpretations.